source/real_locations.rst

```diff
@@ -0,0 +1,116 @@
+
+.. note:: The same thing happens while calculating ``infectedNeighbourCount``. This has effects not just on the workplace, but on homes, hospitals, and other locations in your model too.
+
+Dealing with the problem
+------------------------
+
+There are two currently proposed methods:
+
+Using an attribute of the ``Person`` class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We can solve the problem by adding an attribute called ``currentLocation`` to the ``Person`` class.
+
+.. tip:: Remember to also initialize the attribute when created an instance of the class while extracting data from the CSV!
+
+After doing so, we need to add a behaviour which changes the ``currentLocation`` at every tick. We could do this by adding the following line of code in the ``Person`` class:
+
+.. code-block:: scala
+
+  private val checkCurrentLocation: Context => Unit = (context: Context) => {
+    val schedule = context.fetchScheduleFor(this).get
+    val locationNextTick: String = schedule.getForStep(context.getCurrentStep + 1)
+    if (currentLocation != locationNextTick) {
+      this.updateParam("currentLocation", locationNextTick)
+    }
+  }
+  addBehaviour(checkCurrentLocation)
+
+.. hint:: ``updateParam`` only updates the value of the attribute at the **end** of the tick. Thus, for all practical purposes, it's useful to view the function as one that changes the value of the attribute on the *subsequent tick*. As such, we store the place the person is expected to be on the next tick, and hencee use ``context.getCurrentStep+1`` as an argument to ``schedule.getForStep``.
+
+Now, we need use this attribute when we compute ``totalNeighbourCount`` and ``infectedNeighbourCount``. ``node.getConnectionCount`` has another (optional) argument besides the relation, which is ``matchPattern``. Using it, we can get counts of the people with the relationship who's attributes follow a pattern: in this case, we'll look for the people who have the ``currentLocation`` attribute equal to the ``placeType`` of the node.
+
+
+.. code-block:: scala
+
+  def computeInfectedFraction(node: Node, placeType: String, context: Context): Double = {
+    val totalNeighbourCount = node.getConnectionCount(node.getRelation[Person]().get,
+      "currentLocation" equ placeType)
+
+    if (totalNeighbourCount == 0) return 0d
+
+    val infectedNeighbourCount = node.getConnectionCount(node.getRelation[Person]().get,
+      ("infectionState" equ Infected) and ("currentLocation" equ placeType))
+
+    infectedNeighbourCount.toDouble / totalNeighbourCount.toDouble
+  }
+
+.. note:: ``equ``, ``and`` and other pattern-matching relations are defined in ``import com.bharatsim.engine.graph.patternMatcher.MatchCondition._``
+
+Checking the locations without a parameter
```

Comment:
"parameter" is too generic. Why not say "without using a ``currentLocation`` attribute?"